src/components/Pages/Event/EventDetailCard/index.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { useIntl } from 'react-intl';
+import moment from 'moment-timezone';
+import { events as i18n, locations as i18n2 } from 'js/i18n/definitions';
+
+import EventLocationDetail from 'components/Pages/Event/EventDetailCard/EventLocationDetail';
+import EventDetailFees from 'components/Pages/Event/EventDetailCard/EventDetailFees';
+
+const EventTime = ({ startTime, endTime }) => {
+  if (startTime) {
+    return (
+      <div className="coa-EventDetailItem__time">
+        {startTime && endTime
+          ? `${moment(startTime, 'HH:mm:ss').format('LT')}—${moment(
+              endTime,
+              'HH:mm:ss',
+            ).format('LT')}`
+          : moment(startTime, 'HH:mm:ss').format('LT')}
+      </div>
+    );
+  }
+
+  return null;
+};
+
+const EventDetailCard = ({
+  date,
+  startTime,
+  endTime,
+  location,
+  eventIsFree,
+  fees,
+  registrationUrl,
+}) => {
+  const intl = useIntl();
+  moment.locale(intl.locale);
+
+  return (
+    <div className="coa-EventPage__EventDetailCard">
+      <div className="coa-EventDetailItem">
+        <i className="material-icons">event</i>
+        <div className="coa-EventDetailItem__content">
+          <div className="coa-EventDetailItem__date">
+            {moment(date, 'YYYY-MM-DD').format('dddd • LL')}
+          </div>
+          <EventTime startTime={startTime} endTime={endTime} />
+        </div>
+      </div>
+      {location && location.locationType === 'city_location' ? (
+        <EventLocationDetail
+          name={location.cityLocation.title}
+          street={location.cityLocation.physicalStreet}
+          city={location.cityLocation.physicalCity}
+          state={location.cityLocation.physicalState}
+          zip={location.cityLocation.physicalZip}
+          unit={location.cityLocation.physicalUnit}
+          additionalDetails={location.additionalDetails}
+        />
+      ) : location && location.locationType === 'remote_location' ? (
+        <EventLocationDetail
+          name={location.remoteLocation.name}
+          street={location.remoteLocation.street}
+          city={location.remoteLocation.city}
+          state={location.remoteLocation.state}
+          zip={location.remoteLocation.zip}
+          unit={location.remoteLocation.unit}
+          additionalDetails={location.additionalDetails}
+        />
+      ) : null}
```

Comment:
yeah, that's definitely a way we can do that, I'm pretty sure I just added the check to both when I ran into an error with an event missing a location.

the only thing missing from your example that comes to mind is an explicit check for `locationType == remote_location`

if we add that check back it we'd end up with:
```
location && 
  (location.locationType == city ?
     city code : 
    (location.locationType == remote ? 
      remote code :
      null)
  )
```
instead of the current
```
  location && location.locationType == city ?
     city code : 
    location && location.locationType == remote ? 
      remote code :
      null)
```

Reply:
you're right -- I was thinking we could avoid explicitly checking for remote_location, since there are only two options anyway. I rescind my refactor!

Reply:
alright, let's use reactions to vote on the actual code (jsprettier formatted):

### love it as is ❤️
```
      {location && location.locationType === 'city_location' ? (
        <EventLocationDetail
          name={location.cityLocation.title}
          street={location.cityLocation.physicalStreet}
          city={location.cityLocation.physicalCity}
          state={location.cityLocation.physicalState}
          zip={location.cityLocation.physicalZip}
          unit={location.cityLocation.physicalUnit}
          additionalDetails={location.additionalDetails}
        />
      ) : location && location.locationType === 'remote_location' ? (
        <EventLocationDetail
          name={location.remoteLocation.name}
          street={location.remoteLocation.street}
          city={location.remoteLocation.city}
          state={location.remoteLocation.state}
          zip={location.remoteLocation.zip}
          unit={location.remoteLocation.unit}
          additionalDetails={location.additionalDetails}
        />
      ) : null}
```

### refactor rocketship 🚀

```
      {!!location && location.locationType === 'city_location' ? (
        <EventLocationDetail
          name={location.cityLocation.title}
          street={location.cityLocation.physicalStreet}
          city={location.cityLocation.physicalCity}
          state={location.cityLocation.physicalState}
          zip={location.cityLocation.physicalZip}
          unit={location.cityLocation.physicalUnit}
          additionalDetails={location.additionalDetails}
        />
      ) : location.locationType === 'remote_location' ? (
        <EventLocationDetail
          name={location.remoteLocation.name}
          street={location.remoteLocation.street}
          city={location.remoteLocation.city}
          state={location.remoteLocation.state}
          zip={location.remoteLocation.zip}
          unit={location.remoteLocation.unit}
          additionalDetails={location.additionalDetails}
        />
      ) : null}
```

Reply:
actually, I'm voting for a third option, just a sec